codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -252,13 +251,19 @@ class ServerHttpBoundProtocolTraitImplGenerator(
                             .await
                             .map_err(Into::into)
                     };
+                    use #{FuturesUtil}::future::FutureExt;
```

Comment:
Can we use `TryFutureExt::map_err`?

Reply:
Nice, TIL about `TryFutureExt::map_err`. I switched over to that in https://github.com/awslabs/smithy-rs/pull/2597/commits/e61ed468c90299e485a4c62495b9337922de048d.